src/coreclr/tools/aot/ILCompiler.RyuJit/JitInterface/CorInfoImpl.RyuJit.cs

```diff
@@ -1159,6 +1161,16 @@ private CorInfoHelpFunc getNewArrHelper(CORINFO_CLASS_STRUCT_* arrayCls)
 
             Debug.Assert(type.IsArray);
 
+            TypeDesc elementType = ((ArrayType)type).ElementType;
+
+            if (elementType.IsObject ||
+                elementType.IsInterface ||
+                elementType.IsPointer ||
+                elementType.IsFunctionPointer)
```

Comment:
This looks different from how CoreCLR VM decides this. Is it intentional?

The CoreCLR-equivalent code would be `elementType.IsGCPointer` (this misses `class` and adds unmanaged pointer arrays).

Reply:
As implemented currently, the helper will work for any pointer-sized element, so we can push it to the limit and just check for element size?

Reply:
> Is it intentional?

It was intentional to enable it for more pointer-sized types.

> ...so we can push it to the limit and just check for element size?

I was considering it. If you're fine with it I can change it to that.

Reply:
It is fine with me. We may want to rename the helper to `CORINFO_HELP_NEWARR_1_PTR` and talk about in comments as pointer sized elements.